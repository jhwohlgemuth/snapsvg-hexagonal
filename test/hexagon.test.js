'use strict';

const {mockHexagon} = require('./mocks');
const Hexagon = require('../lib/hexagon');

Object.assign(Hexagon.prototype, mockHexagon);

describe('Hexagon', function() {
    let hexagon;
    afterEach(() => {
        hexagon.addClass.mockClear();
        hexagon.polygon.mockClear();
    });
    it('can be created with default x and y', () => {
        hexagon = new Hexagon(100, 0, 0);
        expect(hexagon.addClass.mock.calls).toMatchSnapshot();
        expect(hexagon.polygon.mock.calls).toMatchSnapshot();
        expect(hexagon).toEqual({});
    });
    it('can be created with correct points', () => {
        hexagon = new Hexagon(100, 10, 10);
        expect(hexagon.addClass.mock.calls).toMatchSnapshot();
        expect(hexagon.polygon.mock.calls).toMatchSnapshot();
        expect(hexagon).toEqual({});
    });
});
