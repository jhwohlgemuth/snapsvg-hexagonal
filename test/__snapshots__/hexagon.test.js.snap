// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hexagon can be created with correct points 1`] = `
Array [
  Array [
    "hexagon",
  ],
]
`;

exports[`Hexagon can be created with correct points 2`] = `
Array [
  Array [
    Array [
      Array [
        10,
        59.99999999999999,
      ],
      Array [
        96.60254037844388,
        10,
      ],
      Array [
        183.20508075688775,
        59.99999999999999,
      ],
      Array [
        183.20508075688775,
        160,
      ],
      Array [
        96.60254037844388,
        210,
      ],
      Array [
        10,
        160,
      ],
    ],
  ],
]
`;

exports[`Hexagon can be created with default x and y 1`] = `
Array [
  Array [
    "hexagon",
  ],
]
`;

exports[`Hexagon can be created with default x and y 2`] = `
Array [
  Array [
    Array [
      Array [
        0,
        49.99999999999999,
      ],
      Array [
        86.60254037844388,
        0,
      ],
      Array [
        173.20508075688775,
        49.99999999999999,
      ],
      Array [
        173.20508075688775,
        150,
      ],
      Array [
        86.60254037844388,
        200,
      ],
      Array [
        0,
        150,
      ],
    ],
  ],
]
`;
